// @accelbyte/sdk-sessionhistory/dist/index.js
import {
  AcquiringDs,
  AcquiringDsWaitTime,
  AllianceRule,
  BackfillProposal,
  BackfillTicket,
  CanceledMatchmakingTicket,
  ConfigAdmin$,
  ConfigAdminApi,
  Configuration,
  CreatedMatchmakingTicket,
  DsInformation,
  EventMatchHistory,
  ExpiredMatchmakingTicket,
  GameServer,
  GameSession,
  GameSessionDetail,
  GameSessionDetail$,
  GameSessionDetailAdmin$,
  GameSessionDetailAdminApi,
  GameSessionDetailApi,
  GameSessionDetailQueryResponse,
  GameSessionTeam,
  History,
  Match,
  MatchLengthDuration,
  MatchMatchmaking,
  MatchMatchmakingTicket,
  MatchTicket,
  MatchingAlly,
  MatchingParty,
  MatchingRule,
  MatchmakingDetail,
  MatchmakingDetailQueryResponse,
  MatchmakingHistory,
  MatchmakingResult,
  Pagination,
  Party,
  PartyDetail,
  PartyDetailQueryResponse,
  PartyHistory,
  PartyMember,
  PartyMembers,
  PartyTeam,
  PlayerData,
  ProposedProposal,
  SessionConfig,
  SessionConfiguration,
  Team,
  Ticket,
  TicketData,
  TicketDetailQueryResponse,
  TicketObservability,
  TicketObservabilityDetail,
  TicketObservabilityHistory,
  TicketStatus,
  TotalActiveSession,
  TotalPlayerPersession,
  User,
  XRayAcquiringDsQueryResponse,
  XRayAcquiringDsWaitTimeQueryResponse,
  XRayAdmin$,
  XRayAdminApi,
  XRayBulkTicketObservabilityResponse,
  XRayCanceledMatchmakingTicketQueryResponse,
  XRayCreatedMatchmakingTicketQueryResponse,
  XRayExpiredMatchmakingTicketQueryResponse,
  XRayMatchHistorQueryResponse,
  XRayMatchLengthDurationQueryResponse,
  XRayMatchMatchmakingQueryResponse,
  XRayMatchMatchmakingTicketQueryResponse,
  XRayMatchPoolPodTickMatchResponse,
  XRayMatchPoolPodTickQueryResponse,
  XRayMatchPoolPodTickResult,
  XRayMatchPoolPodTickTicketResponse,
  XRayMatchPoolQueryResponse,
  XRayMatchPoolResult,
  XRayMatchTicketHistory,
  XRayMatchTicketHistoryQueryResponse,
  XRayMatchesQueryResponse,
  XRayTicketMatchesResult,
  XRayTicketObservabilityResponse,
  XRayTicketQueryResponse,
  XRayTicketResult,
  XRayTotalActiveSessionQueryResponse,
  XRayTotalPlayerPersessionAvgQueryResponse
} from "./chunk-GZHUCLSA.js";

// src/generated-definitions/Error.ts
import { z } from "zod";
var Error = z.object({
  attributes: z.record(z.string()),
  errorCode: z.number().int(),
  errorMessage: z.string(),
  message: z.string(),
  name: z.string()
});

// src/generated-definitions/XRayBulkTicketObservabilityRequest.ts
import { z as z3 } from "zod";

// src/generated-definitions/XRayTicketObservabilityRequest.ts
import { z as z2 } from "zod";
var XRayTicketObservabilityRequest = z2.object({
  action: z2.string(),
  activeAllianceRule: AllianceRule.nullish(),
  activeMatchingRule: z2.array(MatchingRule).nullish(),
  function: z2.string().nullish(),
  gameMode: z2.string(),
  isBackfillMatch: z2.boolean(),
  isRuleSetFlexed: z2.boolean(),
  iteration: z2.number().int().nullish(),
  matchID: z2.string().nullish(),
  namespace: z2.string(),
  remainingPlayersPerTicket: z2.array(z2.number().int()).nullish(),
  remainingTickets: z2.number().int().nullish(),
  sessionTickID: z2.string(),
  tickID: z2.number().int(),
  ticketID: z2.string(),
  timeToMatchSec: z2.number().nullish(),
  timestamp: z2.string(),
  unbackfillReason: z2.string().nullish(),
  unmatchReason: z2.string().nullish()
});

// src/generated-definitions/XRayBulkTicketObservabilityRequest.ts
var XRayBulkTicketObservabilityRequest = z3.object({ ticketObservabilities: z3.array(XRayTicketObservabilityRequest) });

// package.json
var name = "@accelbyte/sdk-sessionhistory";
var version = "5.0.4";
var author = "AccelByte Inc";

// src/SessionHistory.ts
console.log(`${name}@${version}`);
var apis = {
  ConfigAdminApi,
  GameSessionDetailAdminApi,
  XRayAdminApi,
  GameSessionDetailApi,
  version: () => console.log({
    version,
    name,
    author
  })
};
var SessionHistory = apis;
export {
  AcquiringDs,
  AcquiringDsWaitTime,
  AllianceRule,
  BackfillProposal,
  BackfillTicket,
  CanceledMatchmakingTicket,
  ConfigAdmin$,
  ConfigAdminApi,
  Configuration,
  CreatedMatchmakingTicket,
  DsInformation,
  Error,
  EventMatchHistory,
  ExpiredMatchmakingTicket,
  GameServer,
  GameSession,
  GameSessionDetail,
  GameSessionDetail$,
  GameSessionDetailAdmin$,
  GameSessionDetailAdminApi,
  GameSessionDetailApi,
  GameSessionDetailQueryResponse,
  GameSessionTeam,
  History,
  Match,
  MatchLengthDuration,
  MatchMatchmaking,
  MatchMatchmakingTicket,
  MatchTicket,
  MatchingAlly,
  MatchingParty,
  MatchingRule,
  MatchmakingDetail,
  MatchmakingDetailQueryResponse,
  MatchmakingHistory,
  MatchmakingResult,
  Pagination,
  Party,
  PartyDetail,
  PartyDetailQueryResponse,
  PartyHistory,
  PartyMember,
  PartyMembers,
  PartyTeam,
  PlayerData,
  ProposedProposal,
  SessionConfig,
  SessionConfiguration,
  SessionHistory,
  Team,
  Ticket,
  TicketData,
  TicketDetailQueryResponse,
  TicketObservability,
  TicketObservabilityDetail,
  TicketObservabilityHistory,
  TicketStatus,
  TotalActiveSession,
  TotalPlayerPersession,
  User,
  XRayAcquiringDsQueryResponse,
  XRayAcquiringDsWaitTimeQueryResponse,
  XRayAdmin$,
  XRayAdminApi,
  XRayBulkTicketObservabilityRequest,
  XRayBulkTicketObservabilityResponse,
  XRayCanceledMatchmakingTicketQueryResponse,
  XRayCreatedMatchmakingTicketQueryResponse,
  XRayExpiredMatchmakingTicketQueryResponse,
  XRayMatchHistorQueryResponse,
  XRayMatchLengthDurationQueryResponse,
  XRayMatchMatchmakingQueryResponse,
  XRayMatchMatchmakingTicketQueryResponse,
  XRayMatchPoolPodTickMatchResponse,
  XRayMatchPoolPodTickQueryResponse,
  XRayMatchPoolPodTickResult,
  XRayMatchPoolPodTickTicketResponse,
  XRayMatchPoolQueryResponse,
  XRayMatchPoolResult,
  XRayMatchTicketHistory,
  XRayMatchTicketHistoryQueryResponse,
  XRayMatchesQueryResponse,
  XRayTicketMatchesResult,
  XRayTicketObservabilityRequest,
  XRayTicketObservabilityResponse,
  XRayTicketQueryResponse,
  XRayTicketResult,
  XRayTotalActiveSessionQueryResponse,
  XRayTotalPlayerPersessionAvgQueryResponse
};
