// @accelbyte/sdk-sessionbrowser/dist/index.js
import {
  ActiveCustomGameResponse,
  ActiveMatchmakingGameResponse,
  AddPlayerResponse,
  AdminSessionResponse,
  CountActiveSessionResponse,
  CustomGameResponse,
  GameSession,
  GameSessionSetting,
  GetSessionHistoryDetailedResponseItem,
  GetSessionHistoryDetailedResponseItemArray,
  GetSessionHistorySearchResponseItemV2,
  GetSessionHistorySearchResponseV2,
  MatchAttributes,
  MatchMaking,
  MatchingAlly,
  MatchingParty,
  Pagination,
  PagingCursor,
  PartyMember,
  RecentPlayerHistory,
  RecentPlayerQueryResponse,
  Server,
  Session$,
  SessionAdmin$,
  SessionAdminApi,
  SessionApi,
  SessionByUserIDsResponse,
  SessionPlayerHistory,
  SessionPlayerJoining,
  SessionQueryResponse,
  SessionResponse,
  StatusHistory
} from "./chunk-S7Z77FEW.js";

// src/generated-definitions/AddPlayerRequest.ts
import { z } from "zod";
var AddPlayerRequest = z.object({ as_spectator: z.boolean(), user_id: z.string() });

// src/generated-definitions/CreateSessionRequest.ts
import { z as z2 } from "zod";
var CreateSessionRequest = z2.object({
  game_session_setting: GameSessionSetting,
  game_version: z2.string(),
  namespace: z2.string(),
  session_type: z2.string(),
  username: z2.string()
});

// src/generated-definitions/Error.ts
import { z as z3 } from "zod";
var Error = z3.object({ errorCode: z3.number().int(), errorMessage: z3.string() });

// src/generated-definitions/ErrorResponseV2.ts
import { z as z4 } from "zod";
var ErrorResponseV2 = z4.object({
  attributes: z4.record(z4.string()),
  errorCode: z4.number().int(),
  errorMessage: z4.string(),
  message: z4.string(),
  name: z4.string()
});

// src/generated-definitions/ErrorV1.ts
import { z as z5 } from "zod";
var ErrorV1 = z5.object({ errorCode: z5.number().int(), errorMessage: z5.string() });

// src/generated-definitions/JoinGameSessionRequest.ts
import { z as z6 } from "zod";
var JoinGameSessionRequest = z6.object({ password: z6.string() });

// src/generated-definitions/UpdateSessionRequest.ts
import { z as z7 } from "zod";
var UpdateSessionRequest = z7.object({ game_max_player: z7.number().int() });

// src/generated-definitions/UpdateSettingsRequest.ts
import { z as z8 } from "zod";
var UpdateSettingsRequest = z8.any();

// package.json
var name = "@accelbyte/sdk-sessionbrowser";
var version = "5.0.4";
var author = "AccelByte Inc";

// src/SessionBrowser.ts
console.log(`${name}@${version}`);
var apis = {
  SessionAdminApi,
  SessionApi,
  version: () => console.log({
    version,
    name,
    author
  })
};
var SessionBrowser = apis;
export {
  ActiveCustomGameResponse,
  ActiveMatchmakingGameResponse,
  AddPlayerRequest,
  AddPlayerResponse,
  AdminSessionResponse,
  CountActiveSessionResponse,
  CreateSessionRequest,
  CustomGameResponse,
  Error,
  ErrorResponseV2,
  ErrorV1,
  GameSession,
  GameSessionSetting,
  GetSessionHistoryDetailedResponseItem,
  GetSessionHistoryDetailedResponseItemArray,
  GetSessionHistorySearchResponseItemV2,
  GetSessionHistorySearchResponseV2,
  JoinGameSessionRequest,
  MatchAttributes,
  MatchMaking,
  MatchingAlly,
  MatchingParty,
  Pagination,
  PagingCursor,
  PartyMember,
  RecentPlayerHistory,
  RecentPlayerQueryResponse,
  Server,
  Session$,
  SessionAdmin$,
  SessionAdminApi,
  SessionApi,
  SessionBrowser,
  SessionByUserIDsResponse,
  SessionPlayerHistory,
  SessionPlayerJoining,
  SessionQueryResponse,
  SessionResponse,
  StatusHistory,
  UpdateSessionRequest,
  UpdateSettingsRequest
};
