// @accelbyte/sdk-basic/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name2 in all)
    __defProp(target, name2, { get: all[name2], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AddCountryGroupRequest: () => AddCountryGroupRequest,
  AddCountryGroupResponse: () => AddCountryGroupResponse,
  AnonymizationAdmin$: () => AnonymizationAdmin$,
  AnonymizationAdminApi: () => AnonymizationAdminApi,
  Basic: () => Basic,
  ConfigAdmin$: () => ConfigAdmin$,
  ConfigAdminApi: () => ConfigAdminApi,
  ConfigCreate: () => ConfigCreate,
  ConfigInfo: () => ConfigInfo,
  ConfigUpdate: () => ConfigUpdate,
  CountryGroupObject: () => CountryGroupObject,
  CountryObject: () => CountryObject,
  CountryObjectArray: () => CountryObjectArray,
  ErrorEntity: () => ErrorEntity,
  FieldValidationError: () => FieldValidationError,
  FileUpload$: () => FileUpload$,
  FileUploadAdmin$: () => FileUploadAdmin$,
  FileUploadAdminApi: () => FileUploadAdminApi,
  FileUploadApi: () => FileUploadApi,
  FileUploadUrlInfo: () => FileUploadUrlInfo,
  Misc$: () => Misc$,
  MiscAdmin$: () => MiscAdmin$,
  MiscAdminApi: () => MiscAdminApi,
  MiscApi: () => MiscApi,
  Namespace$: () => Namespace$,
  NamespaceAdmin$: () => NamespaceAdmin$,
  NamespaceAdminApi: () => NamespaceAdminApi,
  NamespaceApi: () => NamespaceApi,
  NamespaceContext: () => NamespaceContext,
  NamespaceCreate: () => NamespaceCreate,
  NamespaceInfo: () => NamespaceInfo,
  NamespaceInfoArray: () => NamespaceInfoArray,
  NamespacePublisherInfo: () => NamespacePublisherInfo,
  NamespaceSimpleInfo: () => NamespaceSimpleInfo,
  NamespaceStatusUpdate: () => NamespaceStatusUpdate,
  NamespaceUpdate: () => NamespaceUpdate,
  Permission: () => Permission,
  RetrieveCountryGroupResponse: () => RetrieveCountryGroupResponse,
  RetrieveCountryGroupResponseArray: () => RetrieveCountryGroupResponseArray,
  RetrieveTimeResponse: () => RetrieveTimeResponse,
  UpdateCountryGroupRequest: () => UpdateCountryGroupRequest,
  UserProfile$: () => UserProfile$,
  UserProfileAdmin: () => UserProfileAdmin,
  UserProfileAdmin$: () => UserProfileAdmin$,
  UserProfileAdminApi: () => UserProfileAdminApi,
  UserProfileApi: () => UserProfileApi,
  UserProfileBulkRequest: () => UserProfileBulkRequest,
  UserProfileCreate: () => UserProfileCreate,
  UserProfileInfo: () => UserProfileInfo,
  UserProfilePrivateCreate: () => UserProfilePrivateCreate,
  UserProfilePrivateInfo: () => UserProfilePrivateInfo,
  UserProfilePublicInfo: () => UserProfilePublicInfo,
  UserProfilePublicInfoArray: () => UserProfilePublicInfoArray,
  UserProfileStatusUpdate: () => UserProfileStatusUpdate,
  UserProfileUpdate: () => UserProfileUpdate,
  UserZipCode: () => UserZipCode,
  UserZipCodeUpdate: () => UserZipCodeUpdate,
  ValidationErrorEntity: () => ValidationErrorEntity
});
module.exports = __toCommonJS(src_exports);

// package.json
var name = "@accelbyte/sdk-basic";
var version = "6.0.0";
var author = "AccelByte Inc";

// src/generated-admin/AnonymizationAdminApi.ts
var import_sdk2 = require("@accelbyte/sdk");

// src/generated-admin/endpoints/AnonymizationAdmin$.ts
var import_sdk = require("@accelbyte/sdk");
var import_zod = require("zod");
var AnonymizationAdmin$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Anonymize user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11501&lt;/li&gt;&lt;/ul&gt;
   */
  deleteAnonymizationProfile_ByUserId(userId) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/anonymization/profiles".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.delete(url, { params });
    return import_sdk.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod.z.unknown(), "z.unknown()");
  }
};

// src/generated-admin/AnonymizationAdminApi.ts
function AnonymizationAdminApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk2.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk2.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function deleteAnonymizationProfile_ByUserId(userId) {
    const $ = new AnonymizationAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.deleteAnonymizationProfile_ByUserId(userId);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Anonymize user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11501&lt;/li&gt;&lt;/ul&gt;
     */
    deleteAnonymizationProfile_ByUserId
  };
}

// src/generated-admin/ConfigAdminApi.ts
var import_sdk4 = require("@accelbyte/sdk");

// src/generated-admin/endpoints/ConfigAdmin$.ts
var import_sdk3 = require("@accelbyte/sdk");
var import_zod3 = require("zod");

// src/generated-definitions/ConfigInfo.ts
var import_zod2 = require("zod");
var ConfigInfo = import_zod2.z.object({
  createdAt: import_zod2.z.string().nullish(),
  key: import_zod2.z.string().nullish(),
  namespace: import_zod2.z.string().nullish(),
  updatedAt: import_zod2.z.string().nullish(),
  value: import_zod2.z.string().nullish()
});

// src/generated-admin/endpoints/ConfigAdmin$.ts
var ConfigAdmin$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Create a config.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: created config&lt;/li&gt;&lt;/ul&gt;
   */
  createConfig(data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/configs".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.post(url, data, { params });
    return import_sdk3.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, ConfigInfo, "ConfigInfo");
  }
  /**
   * Delete a config.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: created config&lt;/li&gt;&lt;/ul&gt;
   */
  deleteConfig_ByConfigKey(configKey) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/configs/{configKey}".replace("{namespace}", this.namespace).replace("{configKey}", configKey);
    const resultPromise = this.axiosInstance.delete(url, { params });
    return import_sdk3.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod3.z.unknown(), "z.unknown()");
  }
  /**
   * Get a config.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: config&lt;/li&gt;&lt;/ul&gt;
   */
  getConfig_ByConfigKey(configKey) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/configs/{configKey}".replace("{namespace}", this.namespace).replace("{configKey}", configKey);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk3.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, ConfigInfo, "ConfigInfo");
  }
  /**
   * Update a config.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: created config&lt;/li&gt;&lt;/ul&gt;
   */
  patchConfig_ByConfigKey(configKey, data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/configs/{configKey}".replace("{namespace}", this.namespace).replace("{configKey}", configKey);
    const resultPromise = this.axiosInstance.patch(url, data, { params });
    return import_sdk3.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, ConfigInfo, "ConfigInfo");
  }
  /**
   * Get a publisher config.&lt;br&gt;It will return a publisher namespace config of the given namespace and key.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: config&lt;/li&gt;&lt;/ul&gt;
   */
  getPublisherConfig_ByConfigKey(configKey) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/publisher/configs/{configKey}".replace("{namespace}", this.namespace).replace("{configKey}", configKey);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk3.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, ConfigInfo, "ConfigInfo");
  }
};

// src/generated-admin/ConfigAdminApi.ts
function ConfigAdminApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk4.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk4.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function createConfig(data) {
    const $ = new ConfigAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createConfig(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function deleteConfig_ByConfigKey(configKey) {
    const $ = new ConfigAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.deleteConfig_ByConfigKey(configKey);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getConfig_ByConfigKey(configKey) {
    const $ = new ConfigAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getConfig_ByConfigKey(configKey);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function patchConfig_ByConfigKey(configKey, data) {
    const $ = new ConfigAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.patchConfig_ByConfigKey(configKey, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getPublisherConfig_ByConfigKey(configKey) {
    const $ = new ConfigAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getPublisherConfig_ByConfigKey(configKey);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Create a config.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: created config&lt;/li&gt;&lt;/ul&gt;
     */
    createConfig,
    /**
     * Delete a config.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: created config&lt;/li&gt;&lt;/ul&gt;
     */
    deleteConfig_ByConfigKey,
    /**
     * Get a config.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: config&lt;/li&gt;&lt;/ul&gt;
     */
    getConfig_ByConfigKey,
    /**
     * Update a config.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: created config&lt;/li&gt;&lt;/ul&gt;
     */
    patchConfig_ByConfigKey,
    /**
     * Get a publisher config.&lt;br&gt;It will return a publisher namespace config of the given namespace and key.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: config&lt;/li&gt;&lt;/ul&gt;
     */
    getPublisherConfig_ByConfigKey
  };
}

// src/generated-admin/FileUploadAdminApi.ts
var import_sdk6 = require("@accelbyte/sdk");

// src/generated-admin/endpoints/FileUploadAdmin$.ts
var import_sdk5 = require("@accelbyte/sdk");

// src/generated-definitions/FileUploadUrlInfo.ts
var import_zod4 = require("zod");
var FileUploadUrlInfo = import_zod4.z.object({
  accessUrl: import_zod4.z.string().nullish(),
  contentType: import_zod4.z.string().nullish(),
  method: import_zod4.z.string().nullish(),
  url: import_zod4.z.string().nullish()
});

// src/generated-admin/endpoints/FileUploadAdmin$.ts
var FileUploadAdmin$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Generate an upload URL for user content. It&#39;s valid for 10 minutes.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11102&lt;/li&gt;&lt;li&gt;&lt;i&gt;Default maximum file count per user&lt;/i&gt;: 10 files&lt;/li&gt;&lt;li&gt;&lt;i&gt;Default maximum file size per user&lt;/i&gt;: 104857600 bytes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: URL data&lt;/li&gt;&lt;/ul&gt;
   */
  createFile_ByUserId(userId, queryParams) {
    const params = { category: "default", ...queryParams };
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/files".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.post(url, null, { params });
    return import_sdk5.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, FileUploadUrlInfo, "FileUploadUrlInfo");
  }
  /**
   * Generate an upload URL. It&#39;s valid for 10 minutes.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11101&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: URL data&lt;/li&gt;&lt;/ul&gt;
   */
  createFile_ByFolder(folder, queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/admin/namespaces/{namespace}/folders/{folder}/files".replace("{namespace}", this.namespace).replace("{folder}", folder);
    const resultPromise = this.axiosInstance.post(url, null, { params });
    return import_sdk5.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, FileUploadUrlInfo, "FileUploadUrlInfo");
  }
};

// src/generated-admin/FileUploadAdminApi.ts
function FileUploadAdminApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk6.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk6.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function createFile_ByUserId(userId, queryParams) {
    const $ = new FileUploadAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createFile_ByUserId(userId, queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function createFile_ByFolder(folder, queryParams) {
    const $ = new FileUploadAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createFile_ByFolder(folder, queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Generate an upload URL for user content. It&#39;s valid for 10 minutes.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11102&lt;/li&gt;&lt;li&gt;&lt;i&gt;Default maximum file count per user&lt;/i&gt;: 10 files&lt;/li&gt;&lt;li&gt;&lt;i&gt;Default maximum file size per user&lt;/i&gt;: 104857600 bytes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: URL data&lt;/li&gt;&lt;/ul&gt;
     */
    createFile_ByUserId,
    /**
     * Generate an upload URL. It&#39;s valid for 10 minutes.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11101&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: URL data&lt;/li&gt;&lt;/ul&gt;
     */
    createFile_ByFolder
  };
}

// src/generated-admin/MiscAdminApi.ts
var import_sdk8 = require("@accelbyte/sdk");

// src/generated-admin/endpoints/MiscAdmin$.ts
var import_sdk7 = require("@accelbyte/sdk");
var import_zod11 = require("zod");

// src/generated-definitions/AddCountryGroupResponse.ts
var import_zod6 = require("zod");

// src/generated-definitions/CountryObject.ts
var import_zod5 = require("zod");
var CountryObject = import_zod5.z.object({ code: import_zod5.z.string().nullish(), name: import_zod5.z.string().nullish() });

// src/generated-definitions/AddCountryGroupResponse.ts
var AddCountryGroupResponse = import_zod6.z.object({
  countries: import_zod6.z.array(CountryObject).nullish(),
  countryGroupCode: import_zod6.z.string().nullish(),
  countryGroupName: import_zod6.z.string().nullish()
});

// src/generated-definitions/CountryGroupObject.ts
var import_zod7 = require("zod");
var CountryGroupObject = import_zod7.z.object({
  countries: import_zod7.z.array(CountryObject).nullish(),
  countryGroupCode: import_zod7.z.string().nullish(),
  countryGroupName: import_zod7.z.string().nullish()
});

// src/generated-definitions/CountryObjectArray.ts
var import_zod8 = require("zod");
var CountryObjectArray = import_zod8.z.array(CountryObject);

// src/generated-definitions/RetrieveCountryGroupResponseArray.ts
var import_zod10 = require("zod");

// src/generated-definitions/RetrieveCountryGroupResponse.ts
var import_zod9 = require("zod");
var RetrieveCountryGroupResponse = import_zod9.z.object({
  countries: import_zod9.z.array(CountryObject).nullish(),
  countryGroupCode: import_zod9.z.string().nullish(),
  countryGroupName: import_zod9.z.string().nullish()
});

// src/generated-definitions/RetrieveCountryGroupResponseArray.ts
var RetrieveCountryGroupResponseArray = import_zod10.z.array(RetrieveCountryGroupResponse);

// src/generated-admin/endpoints/MiscAdmin$.ts
var MiscAdmin$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * @deprecated
   * List countries.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11204&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: country code list&lt;/li&gt;&lt;/ul&gt;
   */
  getMiscCountries(queryParams) {
    const params = { lang: "en", ...queryParams };
    const url = "/basic/v1/admin/namespaces/{namespace}/misc/countries".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk7.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, CountryObjectArray, "CountryObjectArray");
  }
  /**
   * List languages.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11206&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: language list&lt;/li&gt;&lt;/ul&gt;
   */
  getMiscLanguages() {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/misc/languages".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk7.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod11.z.unknown(), "z.unknown()");
  }
  /**
   * List time zones.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11205&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: time zones&lt;/li&gt;&lt;/ul&gt;
   */
  getMiscTimezones() {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/misc/timezones".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk7.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod11.z.unknown(), "z.unknown()");
  }
  /**
   * List country groups. Will return all available country groups if the query param is not specified&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11203&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of country groups&lt;/li&gt;&lt;/ul&gt;
   */
  getMiscCountrygroups(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/admin/namespaces/{namespace}/misc/countrygroups".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk7.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      RetrieveCountryGroupResponseArray,
      "RetrieveCountryGroupResponseArray"
    );
  }
  /**
   * Add a country groups&lt;br&gt;Country code must follow ISO3166-1 alpha-2.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11201&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: newly created country group&lt;/li&gt;&lt;/ul&gt;
   */
  createMiscCountrygroup(data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/misc/countrygroups".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.post(url, data, { params });
    return import_sdk7.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      AddCountryGroupResponse,
      "AddCountryGroupResponse"
    );
  }
  /**
   * Delete a country groups by its country group code. This endpoint usually used for testing purpose to cleanup test data.&lt;br&gt;
   */
  deleteMiscCountrygroup_ByCountryGroupCode(countryGroupCode) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/misc/countrygroups/{countryGroupCode}".replace("{namespace}", this.namespace).replace("{countryGroupCode}", countryGroupCode);
    const resultPromise = this.axiosInstance.delete(url, { params });
    return import_sdk7.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod11.z.unknown(), "z.unknown()");
  }
  /**
   * Update a country groups. The countryGroupCode must be exist beforehand.&lt;br&gt;Valid update behaviour :&lt;br/&gt;- To update &lt;i&gt;countryGroupName&lt;/i&gt; only, do not include &lt;i&gt;countries&lt;/i&gt; key or just specify it with empty array.&lt;br/&gt;- To update &lt;i&gt;countries&lt;/i&gt; only, do not include &lt;i&gt;countryGroupName&lt;/i&gt; key or just specify it with blank value.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11202&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: updated country group&lt;/li&gt;&lt;/ul&gt;
   */
  updateMiscCountrygroup_ByCountryGroupCode(countryGroupCode, data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/misc/countrygroups/{countryGroupCode}".replace("{namespace}", this.namespace).replace("{countryGroupCode}", countryGroupCode);
    const resultPromise = this.axiosInstance.put(url, data, { params });
    return import_sdk7.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, CountryGroupObject, "CountryGroupObject");
  }
};

// src/generated-admin/MiscAdminApi.ts
function MiscAdminApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk8.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk8.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function getMiscCountries(queryParams) {
    const $ = new MiscAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getMiscCountries(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getMiscLanguages() {
    const $ = new MiscAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getMiscLanguages();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getMiscTimezones() {
    const $ = new MiscAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getMiscTimezones();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getMiscCountrygroups(queryParams) {
    const $ = new MiscAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getMiscCountrygroups(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function createMiscCountrygroup(data) {
    const $ = new MiscAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createMiscCountrygroup(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function deleteMiscCountrygroup_ByCountryGroupCode(countryGroupCode) {
    const $ = new MiscAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.deleteMiscCountrygroup_ByCountryGroupCode(countryGroupCode);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function updateMiscCountrygroup_ByCountryGroupCode(countryGroupCode, data) {
    const $ = new MiscAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.updateMiscCountrygroup_ByCountryGroupCode(countryGroupCode, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * @deprecated
     * List countries.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11204&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: country code list&lt;/li&gt;&lt;/ul&gt;
     */
    getMiscCountries,
    /**
     * List languages.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11206&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: language list&lt;/li&gt;&lt;/ul&gt;
     */
    getMiscLanguages,
    /**
     * List time zones.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11205&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: time zones&lt;/li&gt;&lt;/ul&gt;
     */
    getMiscTimezones,
    /**
     * List country groups. Will return all available country groups if the query param is not specified&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11203&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of country groups&lt;/li&gt;&lt;/ul&gt;
     */
    getMiscCountrygroups,
    /**
     * Add a country groups&lt;br&gt;Country code must follow ISO3166-1 alpha-2.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11201&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: newly created country group&lt;/li&gt;&lt;/ul&gt;
     */
    createMiscCountrygroup,
    /**
     * Delete a country groups by its country group code. This endpoint usually used for testing purpose to cleanup test data.&lt;br&gt;
     */
    deleteMiscCountrygroup_ByCountryGroupCode,
    /**
     * Update a country groups. The countryGroupCode must be exist beforehand.&lt;br&gt;Valid update behaviour :&lt;br/&gt;- To update &lt;i&gt;countryGroupName&lt;/i&gt; only, do not include &lt;i&gt;countries&lt;/i&gt; key or just specify it with empty array.&lt;br/&gt;- To update &lt;i&gt;countries&lt;/i&gt; only, do not include &lt;i&gt;countryGroupName&lt;/i&gt; key or just specify it with blank value.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11202&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: updated country group&lt;/li&gt;&lt;/ul&gt;
     */
    updateMiscCountrygroup_ByCountryGroupCode
  };
}

// src/generated-admin/NamespaceAdminApi.ts
var import_sdk10 = require("@accelbyte/sdk");

// src/generated-admin/endpoints/NamespaceAdmin$.ts
var import_sdk9 = require("@accelbyte/sdk");

// src/generated-definitions/NamespaceContext.ts
var import_zod12 = require("zod");
var NamespaceContext = import_zod12.z.object({
  namespace: import_zod12.z.string().nullish(),
  publisherNamespace: import_zod12.z.string().nullish(),
  studioNamespace: import_zod12.z.string().nullish(),
  type: import_zod12.z.enum(["Game", "Publisher", "Studio"]).nullish()
});

// src/generated-definitions/NamespaceInfo.ts
var import_zod13 = require("zod");
var NamespaceInfo = import_zod13.z.object({
  clientId: import_zod13.z.string().nullish(),
  createdAt: import_zod13.z.string().nullish(),
  displayName: import_zod13.z.string(),
  namespace: import_zod13.z.string(),
  parentNamespace: import_zod13.z.string().nullish(),
  status: import_zod13.z.enum(["ACTIVE", "DELETED", "INACTIVE"]).nullish(),
  updatedAt: import_zod13.z.string().nullish()
});

// src/generated-definitions/NamespaceInfoArray.ts
var import_zod14 = require("zod");
var NamespaceInfoArray = import_zod14.z.array(NamespaceInfo);

// src/generated-definitions/NamespacePublisherInfo.ts
var import_zod15 = require("zod");
var NamespacePublisherInfo = import_zod15.z.object({ publisher: import_zod15.z.boolean().nullish(), publisherNamespace: import_zod15.z.string().nullish() });

// src/generated-admin/endpoints/NamespaceAdmin$.ts
var NamespaceAdmin$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Get all namespaces.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11303&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of namespaces&lt;/li&gt;&lt;/ul&gt;
   */
  getNamespaces(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/admin/namespaces";
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfoArray, "NamespaceInfoArray");
  }
  /**
   * Create a namespace.&lt;br&gt;By default the namespace is enabled.&lt;br&gt;In multi tenant mode, parentNamespace will be automatically filled with requester namespace if the requester is using studio or publisher token, and it will be filled with studio namespace if the requester uses game token. An oauth client will also be created and the id will be returned. &lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11301&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: created namespace&lt;/li&gt;&lt;/ul&gt;
   */
  createNamespace(data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces";
    const resultPromise = this.axiosInstance.post(url, data, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfo, "NamespaceInfo");
  }
  /**
   * Delete a namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11307&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: deleted namespace&lt;/li&gt;&lt;/ul&gt;
   */
  deleteNamespace_ByNamespace() {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.delete(url, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfo, "NamespaceInfo");
  }
  /**
   * Get a namespace.&lt;br&gt;In multi tenant mode, parentNamespace will be returned.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11304&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: namespace&lt;/li&gt;&lt;/ul&gt;
   */
  getNamespace_ByNamespace(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/admin/namespaces/{namespace}".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfo, "NamespaceInfo");
  }
  /**
   * Get game namespaces.&lt;br&gt;In multi tenant mode, a given super admin namespace will return all game namespaces of studio namespaces&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11308&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of namespaces&lt;/li&gt;&lt;/ul&gt;
   */
  getGame(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/admin/namespaces/{namespace}/game".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfoArray, "NamespaceInfoArray");
  }
  /**
   * Update namespace basic info.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11302&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: updated namespace&lt;/li&gt;&lt;/ul&gt;
   */
  patchBasic(data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/basic".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.patch(url, data, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfo, "NamespaceInfo");
  }
  /**
   * Get child namespaces.&lt;br&gt;If input namespace is publisher namespace, then it will return its all studio namespace.&lt;br&gt;If input namespace is studio namespace, then it will return its all game namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of child namespaces&lt;/li&gt;&lt;/ul&gt;
   */
  getChild(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/admin/namespaces/{namespace}/child".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfoArray, "NamespaceInfoArray");
  }
  /**
   * Change a namespace status.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;Action code&lt;i&gt;&lt;/i&gt;: 11306&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: updated namespace&lt;/li&gt;&lt;/ul&gt;
   */
  patchStatus(data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/status".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.patch(url, data, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfo, "NamespaceInfo");
  }
  /**
   * Get context of namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: context of namespace&lt;/li&gt;&lt;/ul&gt;
   */
  getContext() {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/context".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk9.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceContext, "NamespaceContext");
  }
  /**
   * Get namespace info related publisher namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11305&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Namespace info related publisher namespace&lt;/li&gt;&lt;/ul&gt;
   */
  getPublisher() {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/publisher".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk9.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      NamespacePublisherInfo,
      "NamespacePublisherInfo"
    );
  }
};

// src/generated-admin/NamespaceAdminApi.ts
function NamespaceAdminApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk10.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk10.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function getNamespaces(queryParams) {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getNamespaces(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function createNamespace(data) {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createNamespace(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function deleteNamespace_ByNamespace() {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.deleteNamespace_ByNamespace();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getNamespace_ByNamespace(queryParams) {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getNamespace_ByNamespace(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getGame(queryParams) {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getGame(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function patchBasic(data) {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.patchBasic(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getChild(queryParams) {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getChild(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function patchStatus(data) {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.patchStatus(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getContext() {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getContext();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getPublisher() {
    const $ = new NamespaceAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getPublisher();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Get all namespaces.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11303&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of namespaces&lt;/li&gt;&lt;/ul&gt;
     */
    getNamespaces,
    /**
     * Create a namespace.&lt;br&gt;By default the namespace is enabled.&lt;br&gt;In multi tenant mode, parentNamespace will be automatically filled with requester namespace if the requester is using studio or publisher token, and it will be filled with studio namespace if the requester uses game token. An oauth client will also be created and the id will be returned. &lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11301&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: created namespace&lt;/li&gt;&lt;/ul&gt;
     */
    createNamespace,
    /**
     * Delete a namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11307&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: deleted namespace&lt;/li&gt;&lt;/ul&gt;
     */
    deleteNamespace_ByNamespace,
    /**
     * Get a namespace.&lt;br&gt;In multi tenant mode, parentNamespace will be returned.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11304&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: namespace&lt;/li&gt;&lt;/ul&gt;
     */
    getNamespace_ByNamespace,
    /**
     * Get game namespaces.&lt;br&gt;In multi tenant mode, a given super admin namespace will return all game namespaces of studio namespaces&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11308&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of namespaces&lt;/li&gt;&lt;/ul&gt;
     */
    getGame,
    /**
     * Update namespace basic info.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11302&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: updated namespace&lt;/li&gt;&lt;/ul&gt;
     */
    patchBasic,
    /**
     * Get child namespaces.&lt;br&gt;If input namespace is publisher namespace, then it will return its all studio namespace.&lt;br&gt;If input namespace is studio namespace, then it will return its all game namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of child namespaces&lt;/li&gt;&lt;/ul&gt;
     */
    getChild,
    /**
     * Change a namespace status.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;Action code&lt;i&gt;&lt;/i&gt;: 11306&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: updated namespace&lt;/li&gt;&lt;/ul&gt;
     */
    patchStatus,
    /**
     * Get context of namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: context of namespace&lt;/li&gt;&lt;/ul&gt;
     */
    getContext,
    /**
     * Get namespace info related publisher namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11305&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Namespace info related publisher namespace&lt;/li&gt;&lt;/ul&gt;
     */
    getPublisher
  };
}

// src/generated-admin/UserProfileAdminApi.ts
var import_sdk12 = require("@accelbyte/sdk");

// src/generated-admin/endpoints/UserProfileAdmin$.ts
var import_sdk11 = require("@accelbyte/sdk");
var import_zod20 = require("zod");

// src/generated-definitions/UserProfileInfo.ts
var import_zod16 = require("zod");
var UserProfileInfo = import_zod16.z.object({
  avatarLargeUrl: import_zod16.z.string().nullish(),
  avatarSmallUrl: import_zod16.z.string().nullish(),
  avatarUrl: import_zod16.z.string().nullish(),
  customAttributes: import_zod16.z.record(import_zod16.z.any()).nullish(),
  dateOfBirth: import_zod16.z.string().nullish(),
  firstName: import_zod16.z.string().nullish(),
  language: import_zod16.z.string().nullish(),
  lastName: import_zod16.z.string().nullish(),
  namespace: import_zod16.z.string().nullish(),
  publicId: import_zod16.z.string().nullish(),
  status: import_zod16.z.enum(["ACTIVE", "INACTIVE"]).nullish(),
  timeZone: import_zod16.z.string().nullish(),
  userId: import_zod16.z.string().nullish(),
  zipCode: import_zod16.z.string().nullish()
});

// src/generated-definitions/UserProfilePrivateInfo.ts
var import_zod17 = require("zod");
var UserProfilePrivateInfo = import_zod17.z.object({
  avatarLargeUrl: import_zod17.z.string().nullish(),
  avatarSmallUrl: import_zod17.z.string().nullish(),
  avatarUrl: import_zod17.z.string().nullish(),
  customAttributes: import_zod17.z.record(import_zod17.z.any()).nullish(),
  dateOfBirth: import_zod17.z.string().nullish(),
  firstName: import_zod17.z.string().nullish(),
  language: import_zod17.z.string().nullish(),
  lastName: import_zod17.z.string().nullish(),
  namespace: import_zod17.z.string().nullish(),
  privateCustomAttributes: import_zod17.z.record(import_zod17.z.any()).nullish(),
  publicId: import_zod17.z.string().nullish(),
  status: import_zod17.z.enum(["ACTIVE", "INACTIVE"]).nullish(),
  timeZone: import_zod17.z.string().nullish(),
  userId: import_zod17.z.string().nullish(),
  zipCode: import_zod17.z.string().nullish()
});

// src/generated-definitions/UserProfilePublicInfoArray.ts
var import_zod19 = require("zod");

// src/generated-definitions/UserProfilePublicInfo.ts
var import_zod18 = require("zod");
var UserProfilePublicInfo = import_zod18.z.object({
  avatarLargeUrl: import_zod18.z.string().nullish(),
  avatarSmallUrl: import_zod18.z.string().nullish(),
  avatarUrl: import_zod18.z.string().nullish(),
  customAttributes: import_zod18.z.record(import_zod18.z.any()).nullish(),
  namespace: import_zod18.z.string().nullish(),
  publicId: import_zod18.z.string().nullish(),
  timeZone: import_zod18.z.string().nullish(),
  userId: import_zod18.z.string().nullish()
});

// src/generated-definitions/UserProfilePublicInfoArray.ts
var UserProfilePublicInfoArray = import_zod19.z.array(UserProfilePublicInfo);

// src/generated-admin/endpoints/UserProfileAdmin$.ts
var UserProfileAdmin$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Admin get user public profile by ids.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Required permission&lt;/i&gt;: resource=&lt;b&gt;&#34;ADMIN:NAMESPACE:{namespace}:PROFILE&#34;&lt;/b&gt;, action=2 &lt;b&gt;(GET)&lt;/b&gt;&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11405&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profiles&lt;/li&gt;&lt;/ul&gt;
   */
  createProfilePublic(data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/profiles/public".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.post(url, data, { params });
    return import_sdk11.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePublicInfoArray,
      "UserProfilePublicInfoArray"
    );
  }
  /**
   * Get user profile by public id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile info&lt;/li&gt;&lt;/ul&gt;
   */
  getProfilesByPublicId(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/admin/namespaces/{namespace}/profiles/byPublicId".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk11.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserProfileInfo, "UserProfileInfo");
  }
  /**
   * Delete user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11407&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Delete user profile&lt;/li&gt;&lt;/ul&gt;
   */
  deleteProfile_ByUserId(userId) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/profiles".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.delete(url, { params });
    return import_sdk11.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePrivateInfo,
      "UserProfilePrivateInfo"
    );
  }
  /**
   * Get user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;/ul&gt;
   */
  getProfiles_ByUserId(userId) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/profiles".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk11.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePrivateInfo,
      "UserProfilePrivateInfo"
    );
  }
  /**
   * Update user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated user profile&lt;/li&gt;&lt;/ul&gt;
   */
  updateProfile_ByUserId(userId, data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/profiles".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.put(url, data, { params });
    return import_sdk11.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePrivateInfo,
      "UserProfilePrivateInfo"
    );
  }
  /**
   * Update user profile status.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11406&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;/ul&gt;
   */
  patchProfileStatus_ByUserId(userId, data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/profiles/status".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.patch(url, data, { params });
    return import_sdk11.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePrivateInfo,
      "UserProfilePrivateInfo"
    );
  }
  /**
   * Get user custom attributes.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: custom attributes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;/ul&gt;
   */
  getProfilesCustomAttributes_ByUserId(userId) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/profiles/customAttributes".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk11.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod20.z.unknown(), "z.unknown()");
  }
  /**
   * Update partially custom attributes tied to the user id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Request body&lt;/i&gt;: allowed format: JSON object&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated custom attributes&lt;/li&gt;&lt;/ul&gt;
   */
  updateProfileCustomAttribute_ByUserId(userId, data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/profiles/customAttributes".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.put(url, data, { params });
    return import_sdk11.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod20.z.unknown(), "z.unknown()");
  }
  /**
   * Get user private custom attributes.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: custom attributes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;/ul&gt;
   */
  getProfilesPrivateCustomAttributes_ByUserId(userId) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/profiles/privateCustomAttributes".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk11.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod20.z.unknown(), "z.unknown()");
  }
  /**
   * Update partially private custom attributes tied to the user id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Request body&lt;/i&gt;: allowed format: JSON object&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated custom attributes&lt;/li&gt;&lt;/ul&gt;
   */
  updateProfilePrivateCustomAttribute_ByUserId(userId, data) {
    const params = {};
    const url = "/basic/v1/admin/namespaces/{namespace}/users/{userId}/profiles/privateCustomAttributes".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.put(url, data, { params });
    return import_sdk11.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod20.z.unknown(), "z.unknown()");
  }
};

// src/generated-admin/UserProfileAdminApi.ts
function UserProfileAdminApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk12.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk12.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function createProfilePublic(data) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createProfilePublic(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getProfilesByPublicId(queryParams) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfilesByPublicId(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function deleteProfile_ByUserId(userId) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.deleteProfile_ByUserId(userId);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getProfiles_ByUserId(userId) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfiles_ByUserId(userId);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function updateProfile_ByUserId(userId, data) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.updateProfile_ByUserId(userId, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function patchProfileStatus_ByUserId(userId, data) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.patchProfileStatus_ByUserId(userId, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getProfilesCustomAttributes_ByUserId(userId) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfilesCustomAttributes_ByUserId(userId);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function updateProfileCustomAttribute_ByUserId(userId, data) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.updateProfileCustomAttribute_ByUserId(userId, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getProfilesPrivateCustomAttributes_ByUserId(userId) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfilesPrivateCustomAttributes_ByUserId(userId);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function updateProfilePrivateCustomAttribute_ByUserId(userId, data) {
    const $ = new UserProfileAdmin$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.updateProfilePrivateCustomAttribute_ByUserId(userId, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Admin get user public profile by ids.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Required permission&lt;/i&gt;: resource=&lt;b&gt;&#34;ADMIN:NAMESPACE:{namespace}:PROFILE&#34;&lt;/b&gt;, action=2 &lt;b&gt;(GET)&lt;/b&gt;&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11405&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profiles&lt;/li&gt;&lt;/ul&gt;
     */
    createProfilePublic,
    /**
     * Get user profile by public id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile info&lt;/li&gt;&lt;/ul&gt;
     */
    getProfilesByPublicId,
    /**
     * Delete user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11407&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Delete user profile&lt;/li&gt;&lt;/ul&gt;
     */
    deleteProfile_ByUserId,
    /**
     * Get user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;/ul&gt;
     */
    getProfiles_ByUserId,
    /**
     * Update user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated user profile&lt;/li&gt;&lt;/ul&gt;
     */
    updateProfile_ByUserId,
    /**
     * Update user profile status.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11406&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;/ul&gt;
     */
    patchProfileStatus_ByUserId,
    /**
     * Get user custom attributes.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: custom attributes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;/ul&gt;
     */
    getProfilesCustomAttributes_ByUserId,
    /**
     * Update partially custom attributes tied to the user id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Request body&lt;/i&gt;: allowed format: JSON object&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated custom attributes&lt;/li&gt;&lt;/ul&gt;
     */
    updateProfileCustomAttribute_ByUserId,
    /**
     * Get user private custom attributes.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: custom attributes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;/ul&gt;
     */
    getProfilesPrivateCustomAttributes_ByUserId,
    /**
     * Update partially private custom attributes tied to the user id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Request body&lt;/i&gt;: allowed format: JSON object&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated custom attributes&lt;/li&gt;&lt;/ul&gt;
     */
    updateProfilePrivateCustomAttribute_ByUserId
  };
}

// src/generated-public/FileUploadApi.ts
var import_sdk14 = require("@accelbyte/sdk");

// src/generated-public/endpoints/FileUpload$.ts
var import_sdk13 = require("@accelbyte/sdk");
var FileUpload$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Generate an upload URL for user content. It&#39;s valid for 10 minutes.&lt;br/&gt;There are 2 kinds of storage limitation per user : maximum file count and maximum file size.&lt;br/&gt;The threshold of those limitations is different between upload category that is used.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11102&lt;/li&gt;&lt;li&gt;&lt;i&gt;Default maximum file count per user&lt;/i&gt;: 10 files&lt;/li&gt;&lt;li&gt;&lt;i&gt;Default maximum file size per user&lt;/i&gt;: 104857600 bytes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: URL data&lt;/li&gt;&lt;/ul&gt;
   */
  createFile_ByUserId(userId, queryParams) {
    const params = { category: "default", ...queryParams };
    const url = "/basic/v1/public/namespaces/{namespace}/users/{userId}/files".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.post(url, null, { params });
    return import_sdk13.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, FileUploadUrlInfo, "FileUploadUrlInfo");
  }
  /**
   * Generate an upload URL. It&#39;s valid for 10 minutes.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11101&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: URL data&lt;/li&gt;&lt;/ul&gt;
   */
  createFile_ByFolder(folder, queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/public/namespaces/{namespace}/folders/{folder}/files".replace("{namespace}", this.namespace).replace("{folder}", folder);
    const resultPromise = this.axiosInstance.post(url, null, { params });
    return import_sdk13.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, FileUploadUrlInfo, "FileUploadUrlInfo");
  }
};

// src/generated-public/FileUploadApi.ts
function FileUploadApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk14.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk14.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function createFile_ByUserId(userId, queryParams) {
    const $ = new FileUpload$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createFile_ByUserId(userId, queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function createFile_ByFolder(folder, queryParams) {
    const $ = new FileUpload$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createFile_ByFolder(folder, queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Generate an upload URL for user content. It&#39;s valid for 10 minutes.&lt;br/&gt;There are 2 kinds of storage limitation per user : maximum file count and maximum file size.&lt;br/&gt;The threshold of those limitations is different between upload category that is used.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11102&lt;/li&gt;&lt;li&gt;&lt;i&gt;Default maximum file count per user&lt;/i&gt;: 10 files&lt;/li&gt;&lt;li&gt;&lt;i&gt;Default maximum file size per user&lt;/i&gt;: 104857600 bytes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: URL data&lt;/li&gt;&lt;/ul&gt;
     */
    createFile_ByUserId,
    /**
     * Generate an upload URL. It&#39;s valid for 10 minutes.&lt;br/&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11101&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: URL data&lt;/li&gt;&lt;/ul&gt;
     */
    createFile_ByFolder
  };
}

// src/generated-public/MiscApi.ts
var import_sdk16 = require("@accelbyte/sdk");

// src/generated-public/endpoints/Misc$.ts
var import_sdk15 = require("@accelbyte/sdk");
var import_zod22 = require("zod");

// src/generated-definitions/RetrieveTimeResponse.ts
var import_zod21 = require("zod");
var RetrieveTimeResponse = import_zod21.z.object({ currentTime: import_zod21.z.string().nullish() });

// src/generated-public/endpoints/Misc$.ts
var Misc$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Get server time
   */
  getMiscTime() {
    const params = {};
    const url = "/basic/v1/public/misc/time";
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk15.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, RetrieveTimeResponse, "RetrieveTimeResponse");
  }
  /**
   * @deprecated
   * List countries.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: country code list&lt;/li&gt;&lt;/ul&gt;
   */
  getMiscCountries(queryParams) {
    const params = { lang: "en", ...queryParams };
    const url = "/basic/v1/public/namespaces/{namespace}/misc/countries".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk15.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, CountryObjectArray, "CountryObjectArray");
  }
  /**
   * List languages.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: language list&lt;/li&gt;&lt;/ul&gt;
   */
  getMiscLanguages() {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/misc/languages".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk15.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod22.z.unknown(), "z.unknown()");
  }
  /**
   * List time zones.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: time zones&lt;/li&gt;&lt;/ul&gt;
   */
  getMiscTimezones() {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/misc/timezones".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk15.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod22.z.unknown(), "z.unknown()");
  }
};

// src/generated-public/MiscApi.ts
function MiscApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk16.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk16.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function getMiscTime() {
    const $ = new Misc$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getMiscTime();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getMiscCountries(queryParams) {
    const $ = new Misc$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getMiscCountries(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getMiscLanguages() {
    const $ = new Misc$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getMiscLanguages();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getMiscTimezones() {
    const $ = new Misc$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getMiscTimezones();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Get server time
     */
    getMiscTime,
    /**
     * @deprecated
     * List countries.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: country code list&lt;/li&gt;&lt;/ul&gt;
     */
    getMiscCountries,
    /**
     * List languages.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: language list&lt;/li&gt;&lt;/ul&gt;
     */
    getMiscLanguages,
    /**
     * List time zones.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: time zones&lt;/li&gt;&lt;/ul&gt;
     */
    getMiscTimezones
  };
}

// src/generated-public/NamespaceApi.ts
var import_sdk18 = require("@accelbyte/sdk");

// src/generated-public/endpoints/Namespace$.ts
var import_sdk17 = require("@accelbyte/sdk");

// src/generated-definitions/NamespaceSimpleInfo.ts
var import_zod23 = require("zod");
var NamespaceSimpleInfo = import_zod23.z.object({ displayName: import_zod23.z.string() });

// src/generated-public/endpoints/Namespace$.ts
var Namespace$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Get all namespaces.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11303&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of namespaces&lt;/li&gt;&lt;/ul&gt;
   */
  getNamespaces(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/public/namespaces";
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk17.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceInfoArray, "NamespaceInfoArray");
  }
  /**
   * Get a namespace info.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: namespace info&lt;/li&gt;&lt;/ul&gt;
   */
  getNamespace_ByNamespace() {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk17.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, NamespaceSimpleInfo, "NamespaceSimpleInfo");
  }
  /**
   * Get namespace info related publisher namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11305&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Namespace info related publisher namespace&lt;/li&gt;&lt;/ul&gt;
   */
  getPublisher() {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/publisher".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk17.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      NamespacePublisherInfo,
      "NamespacePublisherInfo"
    );
  }
};

// src/generated-public/NamespaceApi.ts
function NamespaceApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk18.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk18.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function getNamespaces(queryParams) {
    const $ = new Namespace$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getNamespaces(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getNamespace_ByNamespace() {
    const $ = new Namespace$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getNamespace_ByNamespace();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getPublisher() {
    const $ = new Namespace$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getPublisher();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Get all namespaces.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11303&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: list of namespaces&lt;/li&gt;&lt;/ul&gt;
     */
    getNamespaces,
    /**
     * Get a namespace info.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: namespace info&lt;/li&gt;&lt;/ul&gt;
     */
    getNamespace_ByNamespace,
    /**
     * Get namespace info related publisher namespace.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11305&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Namespace info related publisher namespace&lt;/li&gt;&lt;/ul&gt;
     */
    getPublisher
  };
}

// src/generated-public/UserProfileApi.ts
var import_sdk20 = require("@accelbyte/sdk");

// src/generated-public/endpoints/UserProfile$.ts
var import_sdk19 = require("@accelbyte/sdk");
var import_zod25 = require("zod");

// src/generated-definitions/UserZipCode.ts
var import_zod24 = require("zod");
var UserZipCode = import_zod24.z.object({ zipCode: import_zod24.z.string().nullish() });

// src/generated-public/endpoints/UserProfile$.ts
var UserProfile$ = class {
  // @ts-ignore
  // prettier-ignore
  constructor(axiosInstance, namespace, useSchemaValidation = true) {
    this.axiosInstance = axiosInstance;
    this.namespace = namespace;
    this.useSchemaValidation = useSchemaValidation;
  }
  /**
   * Get user public profile by ids.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11405&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profiles&lt;/li&gt;&lt;/ul&gt;
   */
  getProfilesPublic(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/public/namespaces/{namespace}/profiles/public".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk19.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePublicInfoArray,
      "UserProfilePublicInfoArray"
    );
  }
  /**
   * Bulk get user public profile by ids.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profiles&lt;/li&gt;&lt;/ul&gt;
   */
  createProfilePublic(data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/profiles/public".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.post(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePublicInfoArray,
      "UserProfilePublicInfoArray"
    );
  }
  /**
   * Get my profile&lt;br&gt;&lt;b&gt;Client with user token can get user profile in target namespace&lt;/b&gt;&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;li&gt;&lt;i&gt;Path&#39;s namespace&lt;/i&gt; : &lt;ul&gt;&lt;li&gt;can be filled with &lt;b&gt;publisher namespace&lt;/b&gt; in order to get &lt;b&gt;publisher user profile&lt;/b&gt;&lt;/li&gt;&lt;li&gt;can be filled with &lt;b&gt;game namespace&lt;/b&gt; in order to get &lt;b&gt;game user profile&lt;/b&gt;&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;/ul&gt;
   */
  getUsersMeProfiles() {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/me/profiles".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk19.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePrivateInfo,
      "UserProfilePrivateInfo"
    );
  }
  /**
   * Create my profile.&lt;br&gt;&lt;b&gt;Client with user token can create user profile in target namespace&lt;/b&gt;&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11401&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Created user profile&lt;/li&gt;&lt;li&gt;&lt;i&gt;Path&#39;s namespace&lt;/i&gt; : &lt;ul&gt;&lt;li&gt;can be filled with &lt;b&gt;publisher namespace&lt;/b&gt; in order to create &lt;b&gt;publisher user profile&lt;/b&gt;&lt;/li&gt;&lt;li&gt;can be filled with &lt;b&gt;game namespace&lt;/b&gt; in order to create &lt;b&gt;game user profile&lt;/b&gt;&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Country&lt;/i&gt; : ISO3166-1 alpha-2 two letter, e.g. US &lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;/ul&gt;
   */
  createUserMeProfile(data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/me/profiles".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.post(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePrivateInfo,
      "UserProfilePrivateInfo"
    );
  }
  /**
   * Update my profile.&lt;br&gt;Updates user profile in the target namespace (namespace in the path). If token&#39;s namespace doesn&#39;t match the target namespace, the service automatically maps the token&#39;s user ID into the user ID in the target namespace. The endpoint returns the updated user profile on a successful call.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;li&gt;&lt;i&gt;Path&#39;s namespace&lt;/i&gt; : &lt;ul&gt;&lt;li&gt;can be filled with &lt;b&gt;publisher namespace&lt;/b&gt; in order to update &lt;b&gt;publisher user profile&lt;/b&gt;&lt;/li&gt;&lt;li&gt;can be filled with &lt;b&gt;game namespace&lt;/b&gt; in order to update &lt;b&gt;game user profile&lt;/b&gt;&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;/ul&gt;
   */
  updateUserMeProfile(data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/me/profiles".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.put(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(
      this.useSchemaValidation,
      () => resultPromise,
      UserProfilePrivateInfo,
      "UserProfilePrivateInfo"
    );
  }
  /**
   * Get user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;/ul&gt;
   */
  getProfiles_ByUserId(userId) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/{userId}/profiles".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserProfileInfo, "UserProfileInfo");
  }
  /**
   * Create user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11401&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Created user profile&lt;/li&gt;&lt;/ul&gt;
   */
  createProfile_ByUserId(userId, data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/{userId}/profiles".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.post(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserProfileInfo, "UserProfileInfo");
  }
  /**
   * Update user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated user profile&lt;/li&gt;&lt;/ul&gt;
   */
  updateProfile_ByUserId(userId, data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/{userId}/profiles".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.put(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserProfileInfo, "UserProfileInfo");
  }
  /**
   * Get my zip code.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11407&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user zip code&lt;/li&gt;&lt;/ul&gt;
   */
  getUsersMeProfilesZipCode() {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/me/profiles/zipCode".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserZipCode, "UserZipCode");
  }
  /**
   * Update my zip code.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11408&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user zip code&lt;/li&gt;&lt;/ul&gt;
   */
  patchUserMeProfileZipCode(data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/me/profiles/zipCode".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.patch(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserZipCode, "UserZipCode");
  }
  /**
   * Get user public profile by public id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profile&lt;/li&gt;&lt;/ul&gt;
   */
  getProfilesPublicByPublicId(queryParams) {
    const params = { ...queryParams };
    const url = "/basic/v1/public/namespaces/{namespace}/profiles/public/byPublicId".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserProfilePublicInfo, "UserProfilePublicInfo");
  }
  /**
   * Get user profile public info.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11404&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profile&lt;/li&gt;&lt;/ul&gt;
   */
  getProfilesPublic_ByUserId(userId) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/{userId}/profiles/public".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserProfilePublicInfo, "UserProfilePublicInfo");
  }
  /**
   * Update user profile status.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11406&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;/ul&gt;
   */
  patchProfileStatus_ByUserId(userId, data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/{userId}/profiles/status".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.patch(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, UserProfileInfo, "UserProfileInfo");
  }
  /**
   * Get custom attributes info.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11404&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user custom attributes&lt;/li&gt;&lt;/ul&gt;
   */
  getProfilesCustomAttributes_ByUserId(userId) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/{userId}/profiles/customAttributes".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod25.z.unknown(), "z.unknown()");
  }
  /**
   * Update partially custom attributes tied to user id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Request body&lt;/i&gt; : allowed format: JSON object&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated custom attributes&lt;/li&gt;&lt;/ul&gt;
   */
  updateProfileCustomAttribute_ByUserId(userId, data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/{userId}/profiles/customAttributes".replace("{namespace}", this.namespace).replace("{userId}", userId);
    const resultPromise = this.axiosInstance.put(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod25.z.unknown(), "z.unknown()");
  }
  /**
   * Get my private custom attributes.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: custom attributes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;/ul&gt;
   */
  getUsersMeProfilesPrivateCustomAttributes() {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/me/profiles/privateCustomAttributes".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.get(url, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod25.z.unknown(), "z.unknown()");
  }
  /**
   * Update partially private custom attributes tied to me.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Request body&lt;/i&gt;: allowed format: JSON object&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated custom attributes&lt;/li&gt;&lt;/ul&gt;
   */
  updateUserMeProfilePrivateCustomAttribute(data) {
    const params = {};
    const url = "/basic/v1/public/namespaces/{namespace}/users/me/profiles/privateCustomAttributes".replace("{namespace}", this.namespace);
    const resultPromise = this.axiosInstance.put(url, data, { params });
    return import_sdk19.Validate.validateOrReturnResponse(this.useSchemaValidation, () => resultPromise, import_zod25.z.unknown(), "z.unknown()");
  }
};

// src/generated-public/UserProfileApi.ts
function UserProfileApi(sdk, args) {
  const sdkAssembly = sdk.assembly();
  const namespace = args?.coreConfig?.namespace ?? sdkAssembly.coreConfig.namespace;
  const useSchemaValidation = args?.coreConfig?.useSchemaValidation ?? sdkAssembly.coreConfig.useSchemaValidation;
  let axiosInstance = sdkAssembly.axiosInstance;
  const requestConfigOverrides = args?.axiosConfig?.request;
  const baseURLOverride = args?.coreConfig?.baseURL;
  const interceptorsOverride = args?.axiosConfig?.interceptors ?? [];
  if (requestConfigOverrides || baseURLOverride || interceptorsOverride.length > 0) {
    const requestConfig = import_sdk20.ApiUtils.mergeAxiosConfigs(sdkAssembly.axiosInstance.defaults, {
      ...baseURLOverride ? { baseURL: baseURLOverride } : {},
      ...requestConfigOverrides
    });
    axiosInstance = import_sdk20.Network.create(requestConfig);
    for (const interceptor of interceptorsOverride) {
      if (interceptor.type === "request") {
        axiosInstance.interceptors.request.use(interceptor.onRequest, interceptor.onError);
      }
      if (interceptor.type === "response") {
        axiosInstance.interceptors.response.use(interceptor.onSuccess, interceptor.onError);
      }
    }
  }
  async function getProfilesPublic(queryParams) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfilesPublic(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function createProfilePublic(data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createProfilePublic(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getUsersMeProfiles() {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getUsersMeProfiles();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function createUserMeProfile(data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createUserMeProfile(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function updateUserMeProfile(data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.updateUserMeProfile(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getProfiles_ByUserId(userId) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfiles_ByUserId(userId);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function createProfile_ByUserId(userId, data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.createProfile_ByUserId(userId, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function updateProfile_ByUserId(userId, data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.updateProfile_ByUserId(userId, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getUsersMeProfilesZipCode() {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getUsersMeProfilesZipCode();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function patchUserMeProfileZipCode(data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.patchUserMeProfileZipCode(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getProfilesPublicByPublicId(queryParams) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfilesPublicByPublicId(queryParams);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getProfilesPublic_ByUserId(userId) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfilesPublic_ByUserId(userId);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function patchProfileStatus_ByUserId(userId, data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.patchProfileStatus_ByUserId(userId, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getProfilesCustomAttributes_ByUserId(userId) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getProfilesCustomAttributes_ByUserId(userId);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function updateProfileCustomAttribute_ByUserId(userId, data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.updateProfileCustomAttribute_ByUserId(userId, data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function getUsersMeProfilesPrivateCustomAttributes() {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.getUsersMeProfilesPrivateCustomAttributes();
    if (resp.error) throw resp.error;
    return resp.response;
  }
  async function updateUserMeProfilePrivateCustomAttribute(data) {
    const $ = new UserProfile$(axiosInstance, namespace, useSchemaValidation);
    const resp = await $.updateUserMeProfilePrivateCustomAttribute(data);
    if (resp.error) throw resp.error;
    return resp.response;
  }
  return {
    /**
     * Get user public profile by ids.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11405&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profiles&lt;/li&gt;&lt;/ul&gt;
     */
    getProfilesPublic,
    /**
     * Bulk get user public profile by ids.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profiles&lt;/li&gt;&lt;/ul&gt;
     */
    createProfilePublic,
    /**
     * Get my profile&lt;br&gt;&lt;b&gt;Client with user token can get user profile in target namespace&lt;/b&gt;&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;li&gt;&lt;i&gt;Path&#39;s namespace&lt;/i&gt; : &lt;ul&gt;&lt;li&gt;can be filled with &lt;b&gt;publisher namespace&lt;/b&gt; in order to get &lt;b&gt;publisher user profile&lt;/b&gt;&lt;/li&gt;&lt;li&gt;can be filled with &lt;b&gt;game namespace&lt;/b&gt; in order to get &lt;b&gt;game user profile&lt;/b&gt;&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;/ul&gt;
     */
    getUsersMeProfiles,
    /**
     * Create my profile.&lt;br&gt;&lt;b&gt;Client with user token can create user profile in target namespace&lt;/b&gt;&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11401&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Created user profile&lt;/li&gt;&lt;li&gt;&lt;i&gt;Path&#39;s namespace&lt;/i&gt; : &lt;ul&gt;&lt;li&gt;can be filled with &lt;b&gt;publisher namespace&lt;/b&gt; in order to create &lt;b&gt;publisher user profile&lt;/b&gt;&lt;/li&gt;&lt;li&gt;can be filled with &lt;b&gt;game namespace&lt;/b&gt; in order to create &lt;b&gt;game user profile&lt;/b&gt;&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Country&lt;/i&gt; : ISO3166-1 alpha-2 two letter, e.g. US &lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;/ul&gt;
     */
    createUserMeProfile,
    /**
     * Update my profile.&lt;br&gt;Updates user profile in the target namespace (namespace in the path). If token&#39;s namespace doesn&#39;t match the target namespace, the service automatically maps the token&#39;s user ID into the user ID in the target namespace. The endpoint returns the updated user profile on a successful call.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;li&gt;&lt;i&gt;Path&#39;s namespace&lt;/i&gt; : &lt;ul&gt;&lt;li&gt;can be filled with &lt;b&gt;publisher namespace&lt;/b&gt; in order to update &lt;b&gt;publisher user profile&lt;/b&gt;&lt;/li&gt;&lt;li&gt;can be filled with &lt;b&gt;game namespace&lt;/b&gt; in order to update &lt;b&gt;game user profile&lt;/b&gt;&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;/ul&gt;
     */
    updateUserMeProfile,
    /**
     * Get user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;/ul&gt;
     */
    getProfiles_ByUserId,
    /**
     * Create user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11401&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Created user profile&lt;/li&gt;&lt;/ul&gt;
     */
    createProfile_ByUserId,
    /**
     * Update user profile.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Language&lt;/i&gt; : allowed format: en, en-US&lt;/li&gt;&lt;li&gt;&lt;i&gt;Timezone&lt;/i&gt; : IANA time zone, e.g. Asia/Shanghai&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated user profile&lt;/li&gt;&lt;/ul&gt;
     */
    updateProfile_ByUserId,
    /**
     * Get my zip code.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11407&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user zip code&lt;/li&gt;&lt;/ul&gt;
     */
    getUsersMeProfilesZipCode,
    /**
     * Update my zip code.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11408&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user zip code&lt;/li&gt;&lt;/ul&gt;
     */
    patchUserMeProfileZipCode,
    /**
     * Get user public profile by public id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profile&lt;/li&gt;&lt;/ul&gt;
     */
    getProfilesPublicByPublicId,
    /**
     * Get user profile public info.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11404&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user public profile&lt;/li&gt;&lt;/ul&gt;
     */
    getProfilesPublic_ByUserId,
    /**
     * Update user profile status.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11406&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user profile&lt;/li&gt;&lt;/ul&gt;
     */
    patchProfileStatus_ByUserId,
    /**
     * Get custom attributes info.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11404&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: user custom attributes&lt;/li&gt;&lt;/ul&gt;
     */
    getProfilesCustomAttributes_ByUserId,
    /**
     * Update partially custom attributes tied to user id.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Request body&lt;/i&gt; : allowed format: JSON object&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated custom attributes&lt;/li&gt;&lt;/ul&gt;
     */
    updateProfileCustomAttribute_ByUserId,
    /**
     * Get my private custom attributes.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: custom attributes&lt;/li&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11403&lt;/li&gt;&lt;/ul&gt;
     */
    getUsersMeProfilesPrivateCustomAttributes,
    /**
     * Update partially private custom attributes tied to me.&lt;br&gt;Other detail info: &lt;ul&gt;&lt;li&gt;&lt;i&gt;Action code&lt;/i&gt;: 11402&lt;/li&gt;&lt;li&gt;&lt;i&gt;Request body&lt;/i&gt;: allowed format: JSON object&lt;/li&gt;&lt;li&gt;&lt;i&gt;Returns&lt;/i&gt;: Updated custom attributes&lt;/li&gt;&lt;/ul&gt;
     */
    updateUserMeProfilePrivateCustomAttribute
  };
}

// src/Basic.ts
console.log(`${name}@${version}`);
var apis = {
  NamespaceAdminApi,
  ConfigAdminApi,
  MiscAdminApi,
  UserProfileAdminApi,
  FileUploadAdminApi,
  AnonymizationAdminApi,
  MiscApi,
  NamespaceApi,
  UserProfileApi,
  FileUploadApi,
  version: () => console.log({
    version,
    name,
    author
  })
};
var Basic = apis;

// src/generated-definitions/AddCountryGroupRequest.ts
var import_zod26 = require("zod");
var AddCountryGroupRequest = import_zod26.z.object({
  countries: import_zod26.z.array(CountryObject).nullish(),
  countryGroupCode: import_zod26.z.string().nullish(),
  countryGroupName: import_zod26.z.string().nullish()
});

// src/generated-definitions/ConfigCreate.ts
var import_zod27 = require("zod");
var ConfigCreate = import_zod27.z.object({ key: import_zod27.z.string(), value: import_zod27.z.string() });

// src/generated-definitions/ConfigUpdate.ts
var import_zod28 = require("zod");
var ConfigUpdate = import_zod28.z.object({ value: import_zod28.z.string() });

// src/generated-definitions/ErrorEntity.ts
var import_zod30 = require("zod");

// src/generated-definitions/Permission.ts
var import_zod29 = require("zod");
var Permission = import_zod29.z.object({ action: import_zod29.z.number().int().nullish(), resource: import_zod29.z.string().nullish() });

// src/generated-definitions/ErrorEntity.ts
var ErrorEntity = import_zod30.z.object({
  devStackTrace: import_zod30.z.string().nullish(),
  errorCode: import_zod30.z.number().int(),
  errorMessage: import_zod30.z.string(),
  messageVariables: import_zod30.z.record(import_zod30.z.string()).nullish(),
  requiredPermission: Permission.nullish()
});

// src/generated-definitions/FieldValidationError.ts
var import_zod31 = require("zod");
var FieldValidationError = import_zod31.z.object({
  errorCode: import_zod31.z.string().nullish(),
  errorField: import_zod31.z.string().nullish(),
  errorMessage: import_zod31.z.string().nullish(),
  errorValue: import_zod31.z.string().nullish(),
  messageVariables: import_zod31.z.record(import_zod31.z.string()).nullish()
});

// src/generated-definitions/NamespaceCreate.ts
var import_zod32 = require("zod");
var NamespaceCreate = import_zod32.z.object({ displayName: import_zod32.z.string(), namespace: import_zod32.z.string() });

// src/generated-definitions/NamespaceStatusUpdate.ts
var import_zod33 = require("zod");
var NamespaceStatusUpdate = import_zod33.z.object({ status: import_zod33.z.enum(["ACTIVE", "INACTIVE"]) });

// src/generated-definitions/NamespaceUpdate.ts
var import_zod34 = require("zod");
var NamespaceUpdate = import_zod34.z.object({ displayName: import_zod34.z.string() });

// src/generated-definitions/UpdateCountryGroupRequest.ts
var import_zod35 = require("zod");
var UpdateCountryGroupRequest = import_zod35.z.object({ countries: import_zod35.z.array(CountryObject).nullish(), countryGroupName: import_zod35.z.string().nullish() });

// src/generated-definitions/UserProfileAdmin.ts
var import_zod36 = require("zod");
var UserProfileAdmin = import_zod36.z.object({
  avatarLargeUrl: import_zod36.z.string().nullish(),
  avatarSmallUrl: import_zod36.z.string().nullish(),
  avatarUrl: import_zod36.z.string().nullish(),
  customAttributes: import_zod36.z.record(import_zod36.z.any()).nullish(),
  dateOfBirth: import_zod36.z.string().nullish(),
  firstName: import_zod36.z.string().nullish(),
  language: import_zod36.z.string().nullish(),
  lastName: import_zod36.z.string().nullish(),
  privateCustomAttributes: import_zod36.z.record(import_zod36.z.any()).nullish(),
  status: import_zod36.z.enum(["ACTIVE", "INACTIVE"]).nullish(),
  timeZone: import_zod36.z.string().nullish(),
  zipCode: import_zod36.z.string().nullish()
});

// src/generated-definitions/UserProfileBulkRequest.ts
var import_zod37 = require("zod");
var UserProfileBulkRequest = import_zod37.z.object({ userIds: import_zod37.z.array(import_zod37.z.string()).nullish() });

// src/generated-definitions/UserProfileCreate.ts
var import_zod38 = require("zod");
var UserProfileCreate = import_zod38.z.object({
  avatarLargeUrl: import_zod38.z.string().nullish(),
  avatarSmallUrl: import_zod38.z.string().nullish(),
  avatarUrl: import_zod38.z.string().nullish(),
  customAttributes: import_zod38.z.record(import_zod38.z.any()).nullish(),
  dateOfBirth: import_zod38.z.string().nullish(),
  firstName: import_zod38.z.string().nullish(),
  language: import_zod38.z.string().nullish(),
  lastName: import_zod38.z.string().nullish(),
  timeZone: import_zod38.z.string().nullish()
});

// src/generated-definitions/UserProfilePrivateCreate.ts
var import_zod39 = require("zod");
var UserProfilePrivateCreate = import_zod39.z.object({
  avatarLargeUrl: import_zod39.z.string().nullish(),
  avatarSmallUrl: import_zod39.z.string().nullish(),
  avatarUrl: import_zod39.z.string().nullish(),
  customAttributes: import_zod39.z.record(import_zod39.z.any()).nullish(),
  dateOfBirth: import_zod39.z.string().nullish(),
  firstName: import_zod39.z.string().nullish(),
  language: import_zod39.z.string().nullish(),
  lastName: import_zod39.z.string().nullish(),
  privateCustomAttributes: import_zod39.z.record(import_zod39.z.any()).nullish(),
  timeZone: import_zod39.z.string().nullish()
});

// src/generated-definitions/UserProfileStatusUpdate.ts
var import_zod40 = require("zod");
var UserProfileStatusUpdate = import_zod40.z.object({ status: import_zod40.z.enum(["ACTIVE", "INACTIVE"]) });

// src/generated-definitions/UserProfileUpdate.ts
var import_zod41 = require("zod");
var UserProfileUpdate = import_zod41.z.object({
  avatarLargeUrl: import_zod41.z.string().nullish(),
  avatarSmallUrl: import_zod41.z.string().nullish(),
  avatarUrl: import_zod41.z.string().nullish(),
  customAttributes: import_zod41.z.record(import_zod41.z.any()).nullish(),
  dateOfBirth: import_zod41.z.string().nullish(),
  firstName: import_zod41.z.string().nullish(),
  language: import_zod41.z.string().nullish(),
  lastName: import_zod41.z.string().nullish(),
  privateCustomAttributes: import_zod41.z.record(import_zod41.z.any()).nullish(),
  timeZone: import_zod41.z.string().nullish(),
  zipCode: import_zod41.z.string().nullish()
});

// src/generated-definitions/UserZipCodeUpdate.ts
var import_zod42 = require("zod");
var UserZipCodeUpdate = import_zod42.z.object({ zipCode: import_zod42.z.string() });

// src/generated-definitions/ValidationErrorEntity.ts
var import_zod43 = require("zod");
var ValidationErrorEntity = import_zod43.z.object({
  errorCode: import_zod43.z.number().int(),
  errorMessage: import_zod43.z.string(),
  errors: import_zod43.z.array(FieldValidationError).nullish()
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AddCountryGroupRequest,
  AddCountryGroupResponse,
  AnonymizationAdmin$,
  AnonymizationAdminApi,
  Basic,
  ConfigAdmin$,
  ConfigAdminApi,
  ConfigCreate,
  ConfigInfo,
  ConfigUpdate,
  CountryGroupObject,
  CountryObject,
  CountryObjectArray,
  ErrorEntity,
  FieldValidationError,
  FileUpload$,
  FileUploadAdmin$,
  FileUploadAdminApi,
  FileUploadApi,
  FileUploadUrlInfo,
  Misc$,
  MiscAdmin$,
  MiscAdminApi,
  MiscApi,
  Namespace$,
  NamespaceAdmin$,
  NamespaceAdminApi,
  NamespaceApi,
  NamespaceContext,
  NamespaceCreate,
  NamespaceInfo,
  NamespaceInfoArray,
  NamespacePublisherInfo,
  NamespaceSimpleInfo,
  NamespaceStatusUpdate,
  NamespaceUpdate,
  Permission,
  RetrieveCountryGroupResponse,
  RetrieveCountryGroupResponseArray,
  RetrieveTimeResponse,
  UpdateCountryGroupRequest,
  UserProfile$,
  UserProfileAdmin,
  UserProfileAdmin$,
  UserProfileAdminApi,
  UserProfileApi,
  UserProfileBulkRequest,
  UserProfileCreate,
  UserProfileInfo,
  UserProfilePrivateCreate,
  UserProfilePrivateInfo,
  UserProfilePublicInfo,
  UserProfilePublicInfoArray,
  UserProfileStatusUpdate,
  UserProfileUpdate,
  UserZipCode,
  UserZipCodeUpdate,
  ValidationErrorEntity
});
